File: vcmp/events/checkpoint.py
from .abc import Event

class CheckpointEvent(Event):
    ...

class CheckpointEnteredEvent(CheckpointEvent):
    __fields__ = (
        "checkpointid",
        "id",
    )
    checkpointid: int
    id: int

class CheckpointExitedEvent(CheckpointEvent):
    __fields__ = (
        "checkpointid",
        "id",
    )
    checkpointid: int
    id: int

File: crates/vcmp_bindings/src/events/checkpoint.rs
use crate::PlayerId;

#[derive(Debug, Clone)]
pub struct CheckpointEnteredEvent {
    pub check_point_id: i32,
    pub player_id: PlayerId,
}

impl From<(i32, i32)> for CheckpointEnteredEvent {
    fn from(value: (i32, i32)) -> Self {
        Self {
            check_point_id: value.0,
            player_id: value.1,
        }
    }
}

#[derive(Debug, Clone)]
pub struct CheckpointExitedEvent {
    pub check_point_id: i32,
    pub player_id: PlayerId,
}

impl From<(i32, i32)> for CheckpointExitedEvent {
    fn from(value: (i32, i32)) -> Self {
        Self {
            check_point_id: value.0,
            player_id: value.1,
        }
    }
}
